test/Npgsql.Tests/Types/ArrayTests.cs

```diff
@@ -137,6 +138,82 @@ public async Task NullableIntsCannotBeReadAsNonNullable()
             Assert.That(() => reader.GetValue(0), Throws.Exception.TypeOf<InvalidOperationException>());
         }
 
+        [Test, Description("Checks that PG arrays containing nulls are returned as set via ValueTypeArrayMode.")]
+        [TestCase(ValueTypeArrayNullability.Always)]
+        [TestCase(ValueTypeArrayNullability.Never)]
+        [TestCase(ValueTypeArrayNullability.PerInstance)]
+        public async Task ValueTypeArrayNullabilities(ValueTypeArrayNullability mode)
+        {
+            using var pool = CreateTempPool(new NpgsqlConnectionStringBuilder(ConnectionString){ ValueTypeArrayNullability = mode}, out var connectionString);
+            await using var conn = await OpenConnectionAsync(connectionString);
+            await using var cmd = new NpgsqlCommand("SELECT onedim::int[], twodim::int[][] FROM unnest(" +
```

Comment:
nit: `SELECT * FROM (VALUES ...) x` is probably a bit simpler than unnest, in case we care :)

Reply:
Agreed.
I tend to forget about `VALUES` in the context of `SELECT`. Somehow `VALUES` belongs to `INSERT` statements in my mind.